app/controllers/concerns/audit_history.rb

```diff
@@ -0,0 +1,100 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module AuditHistory
+  extend ActiveSupport::Concern
+
+  private
+
+  def access_control_component_config
+    {
+      related_models: [
+        { class: UserGroup, association: :user_group },
+        { class: Role, association: :role },
+        { class: Collection, association: :collection },
+      ],
+      nested_models: [
+        { class: UserGroupMember, parent_association: :user_group, nested_association: :user_group_members },
+        { class: UserRole, parent_association: :role, nested_association: :user_roles },
+        { class: GrdaWarehouse::GroupViewableEntity, parent_association: :collection, nested_association: :group_viewable_entities },
+      ],
+      include_self: false,
+      excluded_fields: ['updated_at'],
+    }
+  end
+
+  def collection_audit_config
+    {
+      referenced_models: [
+        { class: AccessControl, association: :access_controls },
+        { class: GrdaWarehouse::GroupViewableEntity, association: :group_viewable_entities },
```

Comment:
I don't know if it's worth cleaning up, but you shouldn't need the classes here, you can use something like: 
```
self.class.reflect_on_association(:group_viewable_entities).klass
```
On an instance of a `Collection` to get back the `GrdaWarehouse::GroupViewableEntity` class.